bindings/mnt/pyfiction/include/pyfiction/algorithms/simulation/sidb/energy_distribution.hpp

```diff
@@ -24,17 +26,42 @@ void energy_distribution(pybind11::module& m)
 {
     namespace py = pybind11;
 
-    m.def("energy_distribution", &fiction::energy_distribution<Lyt>, py::arg("input_vec"),
-          DOC(fiction_energy_distribution));
+    m.def("calculate_energy_distribution", &fiction::calculate_energy_distribution<Lyt>,
+          py::arg("charge_distributions"), DOC(fiction_energy_distribution));
 }
 
 }  // namespace detail
 
 inline void energy_distribution(pybind11::module& m)
 {
-    // NOTE be careful with the order of the following calls! Python will resolve the first matching overload!
+    namespace py = pybind11;
+
+    py::class_<fiction::energy_state>(m, "energy_state")
+        .def(py::init<double, uint64_t>(), py::arg("electrostatic_potential_energy"), py::arg("degeneracy"),
+             DOC(fiction_energy_state))
+        .def_readwrite("electrostatic_potential_energy", &fiction::energy_state::electrostatic_potential_energy,
+                       DOC(fiction_energy_state_electrostatic_potential_energy))
+        .def_readwrite("degeneracy", &fiction::energy_state::degeneracy, DOC(fiction_energy_state_degeneracy));
 
-    detail::energy_distribution<py_charge_distribution_surface>(m);
+    py::class_<fiction::energy_distribution>(m, "energy_distribution")
+        .def(py::init<>())
+        .def("get_nth_state", &fiction::energy_distribution::get_nth_state, py::arg("state_index"),
+             DOC(fiction_energy_distribution_get_nth_state))
+        .def("degeneracy_of_given_energy", &fiction::energy_distribution::degeneracy_of_given_energy, py::arg("energy"),
```

Comment:
I know we had this discussion in the past. Sorry for bringing it up again. I'm not a big fan of mentioning the parameters in the function name. Instead of names like `add_two_integers_and_return_an_integer` we want `add`.

Reply:
I know and you are right! Thanks for bringing this up again!